src/main/java/org/broadinstitute/hellbender/tools/exome/ReadCountCollectionUtils.java

```diff
@@ -561,8 +563,51 @@ public static ReadCountCollection removeColumnsWithExtremeMedianCounts(final Rea
         }
     }
 
-    private static long countZeroes(final double[] data) {
-        return DoubleStream.of(data).filter(d -> d == 0.0).count();
+    /**
+     * Remove columns with NaNs and infinities
+     *
+     * @param readCounts
+     * @param logger
+     * @return
+     */
+    public static ReadCountCollection removeColumnsWithBadValues(final ReadCountCollection readCounts,
+                                                                 final Logger logger) {
+        final List<String> columnNames = readCounts.columnNames();
+
+        // Determine kept and dropped column sets.
+        final Set<String> columnsToKeep = new LinkedHashSet<>(readCounts.columnNames().size());
+        final Set<String> columnsToDrop = new HashSet<>();
+        for (int i = 0; i < columnNames.size(); i++) {
+            if (Arrays.stream(readCounts.getColumn(i))
+                    .filter(d -> Double.isNaN(d) || Double.isInfinite(d))
+                    .count() == 0) {
+                columnsToKeep.add(columnNames.get(i));
+            } else {
+                columnsToDrop.add(columnNames.get(i));
+            }
+        }
+
+        // log and drop columns if it applies
+        if (columnsToKeep.isEmpty()) {
+            throw new UserException.BadInput("No column count left after removing those with NaN and infinities");
+        } else if (columnsToKeep.size() == columnNames.size()) {
+            logger.info("No column dropped due to bad values");
+            return readCounts;
+        } else {
+            final double droppedPercentage = ((double)(columnsToDrop.size()) / columnNames.size()) * 100;
+            logger.info(String.format("Some columns dropped (%d out of %d, %.2f%%) as they contained bad values: %s",
+                    columnsToDrop.size(), columnNames.size(), droppedPercentage,
+                    columnsToDrop.stream().collect(Collectors.joining(", ", "[", "]"))));
+            return readCounts.subsetColumns(columnsToKeep);
+        }
+    }
+
+    private static long countZeroes(final double[] data, final boolean roundToInteger) {
```

Comment:
The reason to have the rounding option is for handling with GC corrected raw read counts. It has to be `true` to get rid of spurious fractional read counts between 0 and 1 when dealing with GC corrected raw read counts. It has to be 'false' when dealing with proportional coverage. In the test code, it has to be `false` because the test data is proportional coverage. We will get rid of these in the future when we get rid of proportional coverage altogether.